polartools/diffraction.py

```diff
@@ -0,0 +1,155 @@
+            peak_mod=LorentzianModel()
+        if model=='Voigt':
+            peak_mod=VoigtModel()
+        if model=='PseudoVoigt':
+            peak_mod=PseudoVoigtModel()
+
+    background=LinearModel()
+    mod=peak_mod+background
+    pars = background.make_params(intercept=ydata.min(), slope=0)
+    pars += peak_mod.guess(ydata, x=xdata)
+
+    fit=mod.fit(ydata, pars, x=xdata)
+    print(f'Fitting with {model} model')
+    if output:
+        for key in fit.params:
+            print(key, "=", fit.params[key].value, "+/-", fit.params[key].stderr)
+        plt.plot(xdata,ydata)
+        plt.plot(xdata,fit.best_fit)
+        plt.show()
+    return fit
+
+def fit_series(db,scan_series,model='Gaussian',output=False,var_series=None,normalize=False,positioner='4C Theta',detector='APD',**kwargs): 
+    """
+    Fit series of scan with reflection profile of choice and provide fit parameters as list.
+
+    Uses lmfit (https://lmfit.github.io/lmfit-py/).
+
+
+    Parameters
+    ----------
+    db : database
+        Databroker database. If None, it will attempt to read from csv or spec
+        files.
+    scan_series : int
+        start, stop, step, [start2, stop2, step2, ... ,startn, stopn, stepn] 
+    model:
+        fit model: Gaussian, Lorentian, Voigt, PseidoVoigt
+    output:
+        Output fit parameters and plot data+fit for each scan.
+    var_series:
+        Varying variable for scan series, e.g. SampK (sample temperature), optional.
+    normalize:
+        normalize with monitor.
+    positioner : string
+        Name of the positioner, this needs to be the same as defined in
+        Bluesky.
+    detector : string
+        Detector to be read, needs to be the same name as in Bluesky.
+    monitor : string, optional
+        Name of the monitor detector. The returned scans will be
+        detectors/monitor.
+    kwargs :
+        Passed to `load_table` and 'fit_peak'.
+
+    Returns
+    -------
+    fit : lmfit ModelResult class
+        Contains the fit results. See:
+        https://lmfit.github.io/lmfit-py/model.html#the-modelresult-class
+    """
+
+    if len(scan_series)%3: 
+        raise TypeError(f'expected 3*n={3*(len(scan_series)//3)} arguments, got {len(scan_series)}')
+
+    monitor=kwargs.pop('monitor',None)
+    if normalize==False:
+        monitor=None
+
+    nbp=0
+    for series in range(1,len(scan_series),3):
+        nbp=nbp+int(scan_series[series]-scan_series[series-1])/scan_series[series+1]+1
+    fit_result=[np.zeros(7)]
+    for i in range(int(nbp)-1):
+        fit_result.append(np.zeros(7))
+            
+    index=0   
+    for series in range(1,len(scan_series),3):
+        start=scan_series[series-1]
+        stop=scan_series[series]
+        step=scan_series[series+1]
+        print('Intervals: {} to {} with step {}'.format(start,stop,step))
+        #fitnr=0
+        for scan in range(start,stop+1,step):
+            table=load_table(scan,db,detector=detector,monitor=monitor, **kwargs)
```

Comment:
I think `load_scan` is what you need here. if you do:
```
if var_series:
     x, y, parameter = load_scan(db, scan, positioner,[detector, var_series], monitor=monitor, **kwargs)
     fit_result[index][0] = parameter.mean()
else:
     x, y = load_scan(db, scan, positioner, detector, monitor=monitor, **kwargs)
     fit_result[index][0] = index
```

Reply:
Does not work, since temperature would be normalized to monitor as well.